src/applications/caregivers/components/SubmitError/DownloadLink.jsx

```diff
@@ -1,64 +1,88 @@
 import { submitTransform } from 'applications/caregivers/helpers';
 import { apiRequest } from 'platform/utilities/api';
+import { isValidForm } from 'platform/forms-system/src/js/validation';
+import { createFormPageList } from 'platform/forms-system/src/js/helpers';
 
 const DownLoadLink = ({ form }) => {
   const [PDFLink, setPDFLink] = useState(null);
+  const [isLoading, setLoading] = useState(false);
   const formData = submitTransform(formConfig, form);
   const veteranFullName = form.data.veteranFullName;
+  const pageList = createFormPageList(formConfig);
+  const isFormValid = isValidForm(form, pageList);
 
-  useEffect(
-    () => {
-      apiRequest(
-        `${environment.API_URL}/v0/caregivers_assistance_claims/download_pdf`,
-        {
-          method: 'POST',
-          body: formData,
-          headers: {
-            'Content-Type': 'application/json',
-            'Source-App-Name': 'caregivers-10-10cg-',
-          },
+  const downloadPDF = () => {
+    setLoading(true);
+    apiRequest(
+      `${environment.API_URL}/v0/caregivers_assistance_claims/download_pdf`,
+      {
+        method: 'POST',
+        body: formData,
+        headers: {
+          'Content-Type': 'application/json',
+          'Source-App-Name': 'caregivers-10-10cg-',
         },
-      )
-        .then(response => {
-          return response.blob();
-        })
-        .then(blob => {
-          const url = URL.createObjectURL(blob);
-          setPDFLink(url);
-        });
-    },
-    [formData],
-  );
+      },
+    )
+      .then(response => response.blob())
+      .then(blob => {
+        const url = URL.createObjectURL(blob);
+        setPDFLink(url);
+        setLoading(false);
+        recordEvent({ event: 'caregivers-10-10cg-pdf-download--success' });
+      })
+      .catch(error => {
+        setLoading(false);
+        Sentry.withScope(scope => scope.setExtra('error', error));
+        recordEvent({ event: 'caregivers-10-10cg-pdf--failure' });
+      });
+  };
 
-  return (
-    <div className="vads-u-margin-top--2">
-      <a
-        href={PDFLink}
-        download={`10-10CG_${veteranFullName.first}_${veteranFullName.last}`}
-        target="_blank"
-        rel="noreferrer noopener"
-        aria-label="Download 1010CG filled out PDF form"
-      >
-        <i
-          aria-hidden="true"
-          role="img"
-          className="fas fa-download vads-u-padding-right--1"
-        />
-        Download your completed application
-        <span className="sr-only vads-u-margin-right--0p5">
-          `dated ${moment(Date.now()).format('MMM D, YYYY')}`
-        </span>
-        <dfn>
-          <abbr title="Portable Document Format">(PDF)</abbr>
-        </dfn>
-      </a>
-    </div>
-  );
+  const renderPDFLink = () => {
+    return (
+      <div className="pdf-download-link--loaded vads-u-margin-top--2">
+        <a
+          onClick={() => isFormValid.isValid && downloadPDF()}
```

Comment:
If I'm reading this correctly: When it's not valid, a user will still be able to click and nothing will happen: No PDF download, no notification to the user, and no notification to sentry.

Instead, should we: 
-  **Option 1**: disable this download link until the data is valid.
-  **Option 2**: have a message pop up when a user tries to download with invalid data.

Reply:
@kevinmirc the idea to me was to let the UI show the validation itself its just not going to hit the endpoint now.

Reply:
If there are errors inline, then we should just go with **Option 1**. A user shouldn't really be able to click the button if we know ahead of time that their info is invalid (whether or not the button on-click hook is executed or not).

Reply:
@kevinmirc added ErrorState for PDFLink